src/gateways/createArchiveLoader.ts

```diff
@@ -1,41 +1,45 @@
 import { ArchiveBook, ArchiveContent, ArchivePage } from '../app/content/types';
 import { stripIdVersion } from '../app/content/utils';
+import BOOKS from '../config.books';
 import createCache, { Cache } from '../helpers/createCache';
 import { acceptStatus } from '../helpers/fetch';
 
 interface Options {
-  appUrl?: string;
+  appHost?: string; /* alternate host to report to the app for resolving relative links in content */
+  archiveHost?: string; /* this is prefixed on the url unless archiveOverride is passed */
+  archiveOverride?: string; /* this will for sure be used if passed */
   bookCache?: Cache<string, ArchiveBook>;
   pageCache?: Cache<string, ArchivePage>;
 }
 
-const defaultOptions = (backendUrl: string) => ({
-  appUrl: backendUrl,
+const defaultOptions = () => ({
+  archiveHost: '',
   bookCache: createCache<string, ArchiveBook>({maxRecords: 20}),
   pageCache: createCache<string, ArchivePage>({maxRecords: 20}),
 });
 
-/*
- * appUrl is reported to the app for the resolving of relative assets in the content.
- * there are situatons such as pre-rendering using a local proxy where this is different
- * from the actual url that is fetched from.
- */
-export default (backendUrl: string, options: Options = {}) => {
-  const {pageCache, bookCache, appUrl} = {...defaultOptions(backendUrl), ...options};
+export default (archiveBasePath: string, options: Options = {}) => {
+  const {pageCache, bookCache, appHost, archiveHost, archiveOverride} = {
+    ...defaultOptions(),
+    ...options,
+  };
 
-  const contentUrl = (base: string, ref: string) => `${base}/contents/${ref}.json`;
+  const contentUrlBase = (host: string, bookId: string) => archiveOverride ||
+    `${host}${BOOKS[bookId]?.archiveOverride || archiveBasePath}`;
```

Comment:
The issue here is that `archiveOverride` has to be an absolute URL if no host is passed in, which is the case with the update-archive-version script.